accepted/2020/linking-libraries.md

```diff
@@ -1,170 +1,572 @@
+One solution that would solve most of the issues is to create
+[Source Generators](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/)
+for each supported serializer. This would remove the use of Reflection,
+and would allow the trim tool to more easily see which code must be preserved.
+However, Source Generators are a new feature and are potentially expensive to create.
+As such, it may take some time before they are used broadly.
 
-Examples of potential new feature switches:
-- ResourceManager
-    - ResourceManager has logic that can load types from the resource file. This could be conditionally disabled by a feature switch. It is already disabled when loading a `.resource` file directly from disk, which is not embedded in a resource assembly.
-    - Another possibility would be a switch to make exception/resource messages smaller in the libraries. See https://github.com/dotnet/runtime/issues/34057 for more information.
-- EventSource
-    - On some devices the EventSource logic is unnecessary because it has no place to write the events. We could introduce a feature switch to remove EventSource code/calls.
-
-#### Removing Embedded Descriptor Files
-
-There are places in the .NET Libraries where we use private Reflection to call between assemblies. One reason for this is layering. If Assembly A references Assembly B, and Assembly B needs to call a method in Assembly A, it can't reference it directly. We've solved that with a Reflection call.
-
-In order to ensure the linker doesn't remove the private method, we've added "Embedded Descriptor Files" (or `ILLinkTrim.xml` in the `dotnet/runtime` repo) which get embedded into the built assembly. These tell the linker not to trim the specified method/type. The problem is that the linker never trims the specified method/type, even if the callsite of the Reflection is not needed.
-
-See https://github.com/dotnet/runtime/issues/31712 and https://github.com/dotnet/runtime/issues/35199 for more information on this issue.
+### Smaller Size
 
-To solve this, we will follow the approaches outlined in https://github.com/dotnet/runtime/issues/31712 to not rely on the Embedded Descriptor Files in order to make the Reflection usages linker safe. Instead, we will add the appropriate linker annotations and not embed these descriptor files into our libraries. This will allow applications to safely call, or not call, the APIs and the linker will either preserve or trim the code correctly.
+To make applications smaller, we have to let the trim tool know what code is not
+necessary and can be removed. A lot of times it is obvious that if a method
+isn't called, it can be trimmed and the trim tool will recognize this automatically.
+But there are places where unnecessary code looks like it is being called.
+
+#### Removing features
```

Comment:
As with the earlier bulleted list, the previous paragraph reads to me as if it's going to be followed by a list of places where unnecessary code doesn't actually get trimmed, but then these headers aren't about such categories of problems but rather about the solutions to such problems.  Consider changing these headers to describe the category of problem that all share the same solution, rather than naming it based on the solution.